scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowFunctions.scala

```diff
@@ -306,77 +306,84 @@ class TFExampleSCollectionFunctions[T <: Example](val self: SCollection[T]) {
                              compression = compression)
 
   /**
-   * Save this SCollection of `org.tensorflow.example.Example` as a TensorFlow TFRecord file,
-   * along with a protobuf file representing the inferred `org.tensorflow.metadata.v0.Schema`.
-   * If you need to alter the schema, use [[inferExampleMetadata]] and pass altered schema to
-   * [[saveAsTfExampleFile()]].
+   * Saves this SCollection of `org.tensorflow.example.Example` as a TensorFlow TFRecord file.
    * @group output
    */
-  def saveAsTfExampleFile(path: String)(implicit ev: T <:< Example)
-  : (Future[Tap[Example]], Future[Tap[Schema]]) = {
+  def saveAsTfExampleFile(path: String)(implicit ev: T <:< Example): Future[Tap[Example]] = {
     this.saveAsTfExampleFile(
       path,
       schema = null,
+      schemaFilename = "_inferred_schema.pb",
+      suffix = ".tfrecords",
+      compression = Compression.UNCOMPRESSED,
+      numShards = 0
+    )(ev)
+  }
+
+  /**
+   * Saves this SCollection of `org.tensorflow.example.Example` as a TensorFlow TFRecord file,
+   * along with  `org.tensorflow.metadata.v0.Schema`.
+   * @group output
+   */
+  def saveAsTfExampleFile(path: String, schema: Schema)(implicit ev: T <:< Example)
+  : Future[Tap[Example]] = {
+    this.saveAsTfExampleFile(
+      path,
+      schema,
       schemaFilename = "_schema.pb",
       suffix = ".tfrecords",
       compression = Compression.UNCOMPRESSED,
       numShards = 0
     )(ev)
   }
 
+  /**
+   * Saves this SCollection of `org.tensorflow.example.Example` as a TensorFlow TFRecord file,
+   * along with  `org.tensorflow.metadata.v0.Schema`.
+   * @return
+   */
   def saveAsTfExampleFile(path: String,
-                          schema: SCollection[Schema],
+                          schema: Schema,
                           schemaFilename: String,
                           suffix: String,
                           compression: Compression,
                           numShards: Int)
-                          (implicit ev: T <:< Example)
-  : (Future[Tap[Example]], Future[Tap[Schema]]) = {
-    val inferedSchema = Option(schema).getOrElse(self.inferExampleMetadata(ev))
+                          (implicit ev: T <:< Example): Future[Tap[Example]] = {
+    require(schemaFilename != null && schemaFilename != "", "schema filename has to be set!")
     val schemaPath = path.replaceAll("\\/+$", "") + "/" + schemaFilename
+    if (schema == null) {
+      // by default if there is no schema provided infer and save schema
+      self.inferExampleMetadata(schemaPath)(ev)
+    } else {
+      // TODO: maybe enforce some schema checks, but at least save the schema along the data
```

Comment:
Should we track this in a separate issue?

Reply:
@andrewsmartin i was thinking of waiting for Google to release the validation tools and then make a decision whether it even makes sense to do it in scio, but if you wanna create an issue, just in case I'm +1.

Reply:
Nope that sounds good to me, let's wait for Google.

Reply:
+1 on removing the `TODO` in favor of an issue pointing to this.

Reply:
+1